### ISSUE TITLE
Submit-Renewal with -AllOrders or -AllAccounts fails to renew orders with invalid status

### ISSUE THREAD

**Issue by jeffmnall** [rmbolger/Posh-ACME#110]
This is an impressive project that's really saved a lot of work for me.  Of course I'm having one issue.

I'm requesting certs and attempting to renew them using Submit-Renewal from the same machine, using the same creds.  I'm running Submit-Renewal -AllAccounts via Task Scheduler.  I cannot get renewals to go through.  I keep receiving this response in my log file:

Refreshing order *.mydomain.com
Thu, 18 Oct 2018 10:15:21 GMT - [D] Response: {
Thu, 18 Oct 2018 10:15:21 GMT - [D]   "status": "invalid",
Thu, 18 Oct 2018 10:15:21 GMT - [D]   "expires": "2018-10-16T14:05:50Z",
Thu, 18 Oct 2018 10:15:21 GMT - [D]   "identifiers": [
Thu, 18 Oct 2018 10:15:21 GMT - [D]     {
Thu, 18 Oct 2018 10:15:21 GMT - [D]       "type": "dns",
Thu, 18 Oct 2018 10:15:21 GMT - [D]       "value": "*.mydomain.com"
Thu, 18 Oct 2018 10:15:21 GMT - [D]     }
Thu, 18 Oct 2018 10:15:21 GMT - [D]   ],
Thu, 18 Oct 2018 10:15:21 GMT - [D]   "authorizations": [
Thu, 18 Oct 2018 10:15:21 GMT - [D]     "https://acme-v02.api.letsencrypt.org/acme/authz/(myauthzvalue)"
Thu, 18 Oct 2018 10:15:21 GMT - [D]   ],
Thu, 18 Oct 2018 10:15:21 GMT - [D]   "finalize": "https://acme-v02.api.letsencrypt.org/acme/finalize/(MyAccountinformation)/(MyAccountInformation)"
Thu, 18 Oct 2018 10:15:21 GMT - [D] }

I'm retrieving new certs using posh-acme with no issue, but I've so far been unsuccessful in getting renewals working.  It's easy enough to request a new cert every few months, but I'd sure like to set things on automatic. ;)

I've replaced sensitive info with placeholders that I'm sure will be obvious to you.  I'm at a complete loss as to what I'm failing to configure.

**Comment by rmbolger** [comment]
Part of the renewal code is getting the current/old order status from the ACME server before creating a new one (mostly to make sure it uses an existing pending order if one still exists). Technically, all a renewal really is is a new order using the same information as the previous order.

It's actually normal for the old orders to become invalid prior to the certs expiring. Cert and order expirations are separate and an invalid order shouldn't be preventing renewal. Is there other error output that you didn't include?

**Comment by jeffmnall** [comment]
I left out the important part!

I would have expected the new certs to be written to the same directory when I did the initial request, but the old certs remain.

Corresponding with the issue above, I would expect to find renewed certs retrieved and written to:
C:\Users\<MyAccount>\AppData\Local\Posh-ACME\acme-v02.api.letsencrypt.org\<MyAcctInfo>\!.mydomain.com

**Comment by rmbolger** [comment]
Ah ha, so it doesn't think it's time for renewal yet......aaaand I think I found the bug that I swear I fixed previously...but perhaps it only got fixed for individual renewals. Let me go double check.

In the mean time, could you try running an individual one that should be ready for renewal and see if it works?

**Comment by jeffmnall** [comment]
I tried a cert that's expiring on 1/7/2019--that's the oldest I've got at the moment.  I've been chasing this problem for the better part of the year, but might have jumped the gun on it this time?  

When I run $results = Submit-Renewal "<mydomain.com>" I receive:

WARNING: Order for acs-test-ci.net is not recommended for renewal yet. Use -Force to override.

..which I suspect is expected because I'm not in the 30 day window yet.  I am seeing this problem within 30 days, but I have to apologize for sending you on a wild goose chase because I don't have a good example.
